<h1>Activity 1 - PALMER PENGUIN DATA ANALYTICS PROJECT</h1> 
<hr>
<h3> Data Analyst: Johnrey Mansilungan</h3>

In [1]:
import pandas as pd
import numpy as np

<h2>Converting Dataset(s) into Dataframe</h2>
<h2>Requirements</h2>
<ul>
    <li>Name of the Dataframes</li>
    <li>Filepath</li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h2>Descriptive Analysis</h2>

In [4]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

<h2>Finding missing datas</h2>

In [5]:
penguins.isnull

<bound method DataFrame.isnull of      rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2    

In [6]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Adelie Dataframe</h3>

In [9]:
adelie_df = penguins[penguins['species']== 'Adelie']
adelie_df

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

<h3>Gentoo Dataframe</h3>

In [10]:
Gentoo_df = penguins[penguins['species']== 'Gentoo']
Gentoo_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

<h3>Chinstrap Dataframe</h3>

In [11]:
Chinstrap_df = penguins[penguins['species']== 'Chinstrap'] 
Chinstrap_df

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [12]:
penguins[(penguins['species']== 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

rowid    species island  bill_length_mm  bill_depth_mm  \
284    285  Chinstrap  Dream            46.0           18.9   
289    290  Chinstrap  Dream            52.0           18.1   
291    292  Chinstrap  Dream            50.5           19.6   
295    296  Chinstrap  Dream            49.2           18.2   
301    302  Chinstrap  Dream            52.0           19.0   
305    306  Chinstrap  Dream            52.8           20.0   
307    308  Chinstrap  Dream            54.2           20.8   
309    310  Chinstrap  Dream            51.0           18.8   
313    314  Chinstrap  Dream            52.0           20.7   
315    316  Chinstrap  Dream            53.5           19.9   
321    322  Chinstrap  Dream            50.8           18.5   
323    324  Chinstrap  Dream            49.0           19.6   
329    330  Chinstrap  Dream            50.7           19.7   
333    334  Chinstrap  Dream            49.3           19.9   
339    340  Chinstrap  Dream            55.8           19.8   
342    343  Chinstrap  Dream            50.8           19.0   

     flipper_length_mm  body_mass_g     sex  year  
284              195.0       4150.0  female  2007  
289              201.0       4050.0    male  2007  
291              201.0       4050.0    male  2007  
295              195.0       4400.0    male  2007  
301              197.0       4150.0    male  2007  
305              205.0       4550.0    male  2008  
307              201.0       4300.0    male  2008  
309              203.0       4100.0    male  2008  
313              210.0       4800.0    male  2008  
315              205.0       4500.0    male  2008  
321              201.0       4450.0    male  2009  
323              212.0       4300.0    male  2009  
329              203.0       4050.0    male  2009  
333              203.0       4050.0    male  2009  
339              207.0       4000.0    male  2009  
342              210.0       4100.0    male  2009

In [13]:
# penguins_sex = penguins[penguins['sex'] == 'male']
# penguins_sex

<h2>FILLING UP NULL VALUES WITH STATISTICALLY ACCEPTED VALUES</h2>

In [14]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [15]:
penguins['bill_length_mm'].mean()

43.9219298245614

In [16]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [17]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [18]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [19]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [20]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [21]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [22]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [23]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [24]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [25]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [26]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>CHANGING DATA TYPES</h2>

In [27]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  
0           181.000000  3750.000000    male  2007  
1           186.000000  3800.000000  female  2007  
2           195.000000  3250.000000  female  2007  
3           200.915205  4201.754386    male  2007  
4           193.000000  3450.000000  female  2007  
..                 ...          ...     ...   ...  
339         207.000000  4000.000000    male  2009  
340         202.000000  3400.000000  female  2009  
341         193.000000  3775.000000    male  2009  
342         210.000000  4100.000000    male  2009  
343         198.000000  3775.000000  female  2009  

[344 rows x 9 columns]

In [28]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['rowid'] = penguins['rowid'].astype('category')
penguins['island'] = penguins['island'].astype('category')

<h1>7 Questions with descriptive analysis</h1>

<ol>
    <li>Which island has the highest penguin population?</li>
    <li>What is the average bill length per specie?</li>
    <li>How does penguin body mass change from 2007 to 2009?</li>
    <li>Which sex is heavier by average?</li>
    <li>Do male penguins have longer bills than females?</li>
    <li>Which island has the heaviest penguins on average?</li>
    <li>Can we predict the penguin's species using bill length, bill depth, flipper length, and body mass?</li>
    <li>Which species has the highest average bill depth on each island?</li>
    <li>What’s the heaviest penguin recorded per species?</li>
    <li>What is the median flipper length per species + island + sex?</li>
    <li>For each island, what’s the most common species?</li>
    <li>IS there a correlation between bill length and flipper length of Gentoo species</li>
    <li>Within each species, identify the lightest penguin</li>
</ol>
<hr>

<h3>1. Which island has the highest penguin population?</h3>

In [29]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>ANALYSIS</h3>

<p>The island Biscoe has the highest population of penguins with 168 while Torgersen has the lowest with 52.</p>
<hr>

<h3>2. What is the average bill length per specie?</h3>

In [30]:
penguins[penguins['species'] == 'Adelie']['bill_length_mm'].mean().round()

39.0

In [31]:
penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'].mean().round()


49.0

In [32]:
penguins[penguins['species'] == 'Gentoo']['bill_length_mm'].mean().round()


47.0

<h3>ANALYSIS</h3>

<p>Chinstrap specie has the biggest average bill length with the average of 49 mm while Adelie has the lowest with the average of 39 mm. Gentoo comes in the middle with 47mm.</p>
<hr>

<h3>3. How does penguin body mass change from 2007 to 2009?</h3>

In [33]:
penguins[penguins['year'] == 2007 ]['body_mass_g'].mean()

4125.24322169059

In [34]:
penguins[penguins['year'] == 2008 ]['body_mass_g'].mean()

4266.666666666667

In [35]:
penguins[penguins['year'] == 2009 ]['body_mass_g'].mean()

4210.222953216375

<h3>ANALYSIS</h3>

<p>Year 2007 has the lowest average weight with an average weight of 4125 grams while there is a significant increase in weight in the year 2008 with an average weight of 4266 grams. There was a slight decline in the year 2009 with an average of 4210 grams.</p>
<hr>

<h3>4. Which sex is heavier by average?</h3>

In [36]:
penguins.groupby('sex')['body_mass_g'].mean()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\1725385282.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('sex')['body_mass_g'].mean()


sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h3>ANALYSIS</h3>

<p>Male penguin species generally is the heaviest specie between both sex with an average weight of 4514.69 grams.</p>
<hr>

<h3>5. Do male penguins have longer bills than females?</h3>

In [37]:
penguins.groupby(['species', 'sex'])['bill_length_mm'].mean().round()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\2469278284.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby(['species', 'sex'])['bill_length_mm'].mean().round()


species    sex   
Adelie     female    37.0
           male      40.0
Chinstrap  female    47.0
           male      51.0
Gentoo     female    46.0
           male      49.0
Name: bill_length_mm, dtype: float64

<h3>ANALYSIS</h3>

<p>Male specie generally has longer bill than female specie.</p>
<hr>

<h3>6. Which island has the heaviest penguins on average?</h3>

In [38]:
penguins.groupby('island')['body_mass_g'].mean().round()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\4236590180.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('island')['body_mass_g'].mean().round()


island
Biscoe       4713.0
Dream        3713.0
Torgersen    3716.0
Name: body_mass_g, dtype: float64

<h3>ANALYSIS</h3>

<p>Biscoe island has the heaviest penguins on average, averaging 4713 grams. Comes next is the Torgersen island with 3716 grams, Dream island comes last with a close gap with Torgersen with 3713 grames average.</p>
<hr>

<h3>7. Can we predict the penguin's species using bill length, bill depth, flipper length, and body mass?</h3>

In [39]:
penguins.groupby('species')[['bill_length_mm', 'bill_depth_mm', 'flipper_length_mm', 'body_mass_g']].mean().round()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\2338959674.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('species')[['bill_length_mm', 'bill_depth_mm', 'flipper_length_mm', 'body_mass_g']].mean().round()


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie               39.0           18.0              190.0       3704.0
Chinstrap            49.0           18.0              196.0       3733.0
Gentoo               47.0           15.0              217.0       5069.0

<h3>ANALYSIS</h3>

<p>We pretty much can predict the spicies just by its vitals. Adelie specie usually has the shortest bill among all species. Gentoo is generally have the longest flipper length and the heaviest specie among all of them, Adelie and Chistrap almost weigh the same but differs heavily on their bill length.</p>
<hr>

<h3>8. Which species has the highest average bill depth on each island?</h3>

In [87]:
penguins.groupby(['island', 'species'])['bill_depth_mm'].mean().round(2).groupby('island').idxmax()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\488580016.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby(['island', 'species'])['bill_depth_mm'].mean().round(2).groupby('island').idxmax()


island
Biscoe          (Biscoe, Adelie)
Dream         (Dream, Chinstrap)
Torgersen    (Torgersen, Adelie)
Name: bill_depth_mm, dtype: object

<h3>ANALYSIS</h3>

<p>Adelie has the highest bill depth in both Biscoe and Torgersen island while it's Chinstrap in the Dream island.</p>
<hr>

<h3>9. What’s the heaviest penguin recorded per species?</h3>

In [64]:
penguins.groupby('species')['body_mass_g'].max()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\2927113222.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('species')['body_mass_g'].max()


species
Adelie       4775.0
Chinstrap    4800.0
Gentoo       6300.0
Name: body_mass_g, dtype: float64

<h3>ANALYSIS</h3>

<p>The heaviest Adelie species weighs 4775 grams, heaviest Chinstrap species weighs 4800 grams, while it's 6300 grams for the heaviest Gentoo species -which was the heaviest among all species.</p>
<hr>

<h3>10.What is the median flipper length per species + island + sex?</h3>

In [68]:
penguins.groupby(['species', 'island', 'sex'])['flipper_length_mm'].median().unstack()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\3222820512.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby(['species', 'island', 'sex'])['flipper_length_mm'].median().unstack()


sex                  female   male
species   island                  
Adelie    Biscoe      187.0  191.0
          Dream       188.0  190.0
          Torgersen   189.0  195.0
Chinstrap Biscoe        NaN    NaN
          Dream       192.0  200.5
          Torgersen     NaN    NaN
Gentoo    Biscoe      212.0  221.0
          Dream         NaN    NaN
          Torgersen     NaN    NaN

<h3>ANALYSIS</h3>

<p></p>
<hr>

<h3>11. For each island, what’s the most common species?</h3>

In [72]:
penguins.groupby('island')['species'].value_counts()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\2408466179.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('island')['species'].value_counts()


island     species  
Biscoe     Gentoo       124
           Adelie        44
           Chinstrap      0
Dream      Chinstrap     68
           Adelie        56
           Gentoo         0
Torgersen  Adelie        52
           Chinstrap      0
           Gentoo         0
Name: count, dtype: int64

<h3>ANALYSIS</h3>

<p>The most common species in Biscoe island is Gentoo with 124 population, Chinstrap for Dream island with 68, while Adelie for Torgesen. With these numbers, we can pretty much identify where each species possibly originated.</p>
<hr>

<h3>12. IS there a correlation between bill length and flipper length of Gentoo species</h3>

In [89]:
penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])

0.6561813407464278

<h3>ANALYSIS</h3>

<p>There is a correlation between bill length and flipper length of Gentoo species. We can pretty much conclude that the longer the bill lenght of a penguin, the longer its flipper.</p>
<hr>

<h3>13. Within each species, identify the lightest penguin</h3>

In [82]:
penguins.groupby('species')['body_mass_g'].min()

C:\Users\Ryzen 7 5700G\AppData\Local\Temp\ipykernel_23220\467692861.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('species')['body_mass_g'].min()


species
Adelie       2850.0
Chinstrap    2700.0
Gentoo       3950.0
Name: body_mass_g, dtype: float64

<h3>ANALYSIS</h3>

<p>The lightest Adelie species weighs 2850 grams, lightest Chinstrap species weighs 2700 grams, while it's 3950 grams for the lightest Gentoo species. We can now know that Gentoo species has the biggest average penguins.</p>
<hr>